snap/validate.go

```diff
@@ -253,13 +253,13 @@ func validateSocketAddrPath(socket *SocketInfo, fieldName string, path string) e
 		return fmt.Errorf("invalid %q: %q should be written as %q", fieldName, path, clean)
 	}
 
-	switch socket.App.DaemonScope {
-	case SystemDaemon:
+	switch {
+	case socket.App.DaemonScope.IsSystemDaemon():
```

Comment:
Not a lover for a switch in bool functions that technically could be all true... I'd still keep using the user and system daemon scopes where the graphical one is just a subcategory of the user case